encoding/kzg/prover/parametrized_prover.go

```diff
@@ -118,7 +118,7 @@ func (g *ParametrizedProver) Encode(inputFr []fr.Element) (*bn254.G1Affine, *bn2
 	if g.Verbose {
 		log.Printf("Total encoding took      %v\n", time.Since(startTime))
 	}
-	return &commit, &lowDegreeCommitment, &lowDegreeProof, kzgFrames, indices, nil
+	return &commit, &lengthCommitment, &lengthProof, kzgFrames, indices, nil
 }
 
 func (g *ParametrizedProver) Commit(polyFr []fr.Element) (bn254.G1Affine, error) {
```

Comment:
Have you checked the fixed base MSM (L175) as suggested by auditor? If it's useful, we may just file a ticket to track it, I think perf is optional

Reply:
it is just not something we know how to do, it is usually part of crypto library, that is why it is suggested to request this feature to gnark. Anup and I will take further look into it